src/Development/IDE/Import/DependencyInformation.hs

```diff
@@ -268,22 +295,49 @@ transitiveDeps DependencyInformation{..} file = do
       IntSet.delete (getFilePathId fileId) .
       IntSet.fromList . map (fst3 . fromVertex) .
       reachable g <$> toVertex (getFilePathId fileId)
-  let transitiveModuleDepIds = filter (\v -> v `IntSet.member` reachableVs) $ map (fst3 . fromVertex) vs
+  let transitiveModuleDepIds =
+        filter (\v -> v `IntSet.member` reachableVs) $ map (fst3 . fromVertex) vs
   let transitivePkgDeps =
           Set.toList $ Set.unions $
           map (\f -> IntMap.findWithDefault Set.empty f depPkgDeps) $
           getFilePathId fileId : transitiveModuleDepIds
-  let transitiveModuleDeps = map (idToPath depPathIdMap . FilePathId) transitiveModuleDepIds
+  let transitiveModuleDeps =
+        map (idToPath depPathIdMap . FilePathId) transitiveModuleDepIds
+  let transitiveNamedModuleDeps =
+        [ NamedModuleDep (idToPath depPathIdMap (FilePathId fid)) mn ml
+        | (fid, ShowableModuleName mn) <- IntMap.toList depModuleNames
+        , let ArtifactsLocation ml = idToPathMap depPathIdMap IntMap.! fid
+        ]
   pure TransitiveDependencies {..}
-  where (g, fromVertex, toVertex) = graphFromEdges (map (\(f, fs) -> (f, f, IntSet.toList fs)) $ IntMap.toList depModuleDeps)
-        vs = topSort g
+  where
+    (g, fromVertex, toVertex) = graphFromEdges (map (\(f, fs) -> (f, f, IntSet.toList fs)) $ IntMap.toList depModuleDeps)
+    vs = topSort g
 
 data TransitiveDependencies = TransitiveDependencies
   { transitiveModuleDeps :: [NormalizedFilePath]
+  , transitiveNamedModuleDeps :: [NamedModuleDep]
   -- ^ Transitive module dependencies in topological order.
   -- The module itself is not included.
   , transitivePkgDeps :: [InstalledUnitId]
   -- ^ Transitive pkg dependencies in unspecified order.
   } deriving (Eq, Show, Generic)
 
 instance NFData TransitiveDependencies
+
+data NamedModuleDep = NamedModuleDep {
+  nmdFilePath :: !NormalizedFilePath,
+  nmdModuleName :: !ModuleName,
+  nmdModLocation :: !ModLocation
+  }
+  deriving Generic
+
+instance Eq NamedModuleDep where
+  a == b = nmdFilePath a == nmdFilePath b
+
+instance NFData NamedModuleDep where
+  rnf NamedModuleDep{..} =
+    rnf nmdFilePath `seq` rnf nmdModuleName `seq` rwhnf nmdModLocation
```

Comment:
Can you add a comment why it's `rwhnf` of `nmdModLocation`?

Reply:
Added (ModLocation lacks an NFData instance)